README.md

```diff
@@ -1,2 +1,130 @@
+
+Clients can subscribe to one or more events. A typical subscription could be:
+
+<pre>    resource "aws_sns_topic_subscription" "my_probation_subscription" {
+    provider      = aws.london
+    topic_arn     = module.probation_offender_events.topic_arn
+    protocol      = "sqs"
+    endpoint      = module.my_queue.sqs_arn
+    filter_policy = "{\"eventType\":[ \"OFFENDER_DETAILS_CHANGED\", \"OFFENDER_ADDRESS_CHANGED\"] }"
+    }
+
+</pre>
+
+The DPS Tech Team will create the queue and subscription on your behalf. If you are using the MoJ Cloud Platform we will need your namespace to send secrets there for your new queue.
+
+## Architecture and design
+
+The service uses a Spring Scheduler to pull the OFFENDER_DELTA table via the **community-api** 
+
+The scheduler will request all records every 10 seconds and process all records found. Multiple instances of this service can run currently since each record is "locked" when retrieved.
+The lock is "released" by either deleting the record or marking it as permanently failed. Any records that have not been processed with 10 minutes is assumed to ready to be processed again. This logic is encapsulated in the **community-api**.
+
+### Running
+
+`localstack` is used to emulate the AWS SQS and Elastic Search service. When running the integration test this will be started automatically. If you want the tests to use an already running version of `localstack` run the tests with the environment `AWS_PROVIDER=localstack`. This has the benefit of running the test quicker without the overhead of starting the `localstack` container.
+
+Any commands in `localstack/setup-sns.sh`will be run when `localstack` starts, so this should contain commands to create the appropriate topic and test queues.
+
+Running all services locally:
+```bash
+docker-compose up 
+```
+
+*Please note the above will not work on a Mac using docker desktop since the docker network host mode is not supported on a Mac*
+
+For a Mac it recommended running all components *except* probation-offender-events (see below) then running probation-offender-events externally:
+
+```bash
+SPRING_PROFILES_ACTIVE=dev ./gradlew bootRun 
+```
+
+Queues and topics will automatically be created when the `localstack` container starts.
+
+Running all services except this application (hence allowing you to run this in the IDE)
+
+```bash
+docker-compose up --scale probation-offender-events=0 
+```
+
+Depending on the speed of your machine when running all services you may need to scale `probation-offender-events=0` until localstack starts. This is a workaround for an issue whereby Spring Boot gives up trying to connect to SNS when the services first starts up.
+
+### Running tests
+
+#### External localstack
+
+`SNS_PROVIDER=localstack ./gradlew test` will override the default behaviour and will expect localstack to already be started externally. In this mode the following services must be started `sqs,sns,es`
```

Comment:
Possibly change this to `./gradlew check` - makes no difference at the moment but is technically the better task to run in case we ever split the tests into multiple tasks.

Reply:
done